src/_pytest/python.py

```diff
@@ -911,6 +912,23 @@ def hasnew(obj: object) -> bool:
     return False
 
 
+@attr.s(auto_attribs=True, eq=False)
+class SafeHashWrapper:
+    obj: Any
+
+    def __eq__(self, other) -> bool:
+        try:
+            res = self.obj == other
+            return bool(res)
+        except Exception:
+            return id(self.obj) == id(other)
```

Comment:
I guess this is equivalent to `return self.obj is other` so can maybe just do that